src/runtime/registry/google-analytics.ts

```diff
@@ -1,32 +1,38 @@
 // WARNING: This file is automatically generated, do not manually modify.
 import { withQuery } from 'ufo'
-import type { GoogleAnalyticsApi } from 'third-party-capital'
+import type { DataLayer } from 'third-party-capital'
 import { defu } from 'defu'
 import { useRegistryScript } from '#nuxt-scripts-utils'
 import type { RegistryScriptInput } from '#nuxt-scripts'
 import { object, string, optional } from '#nuxt-scripts-validator'
 
-export const GoogleAnalyticsOptions = object({ id: string(), dataLayerName: optional(string()) })
-
 declare global {
-  interface Window extends GoogleAnalyticsApi {}
+  interface Window {
+    [key: string]: any
+  }
 }
+export const GoogleAnalyticsOptions = object({ id: string(), l: optional(string()) })
 export type GoogleAnalyticsInput = RegistryScriptInput<typeof GoogleAnalyticsOptions>
 
-export function useScriptGoogleAnalytics<T extends GoogleAnalyticsApi>(_options?: GoogleAnalyticsInput) {
-  _options = defu(_options, { dataLayerName: 'defaultGa' })
-  return useRegistryScript<T, typeof GoogleAnalyticsOptions>(_options?.key || 'googleAnalytics', options => ({
+function use(options: GoogleAnalyticsInput) {
+  return { dataLayer: window[options.l!] as DataLayer,
+    gtag(...args: any) { (window[options.l!] as DataLayer).push(args) } }
```

Comment:
we had this typed before

Reply:
Are you talking about gtag() ? Not anymore with tpc 2.0. So it's better to rely on our own use function

Reply:
ye gtag sorry, I agree it's better but can we re-add the types

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/gtag.js/index.d.ts

Reply:
`DataLayer` type from tpc conflict with `@types/gtag.js` GtagCommand types. 

cc @flashdesignory

Reply:
@huang-julien - I'm happy to adjust the DataLayer type as needed. Do you want to open a pr or what's the easiest / fastest to get you what you need?

Reply:
@huang-julien - not sure exactly where the conflict is happening, but could you use the GTag type from tpc?

maybe something like:
```
import type { DataLayer, GTag } from 'third-party-capital'

function use(options: GoogleAnalyticsInput) {
  const gtag: GTag = function (...args: any) {
    ((window as any)[options.l ?? 'dataLayer'] as DataLayer)?.push(args)
  }
  return { dataLayer: (window as any)[options.l!] as DataLayer,
    gtag,
  }
}
```